### ISSUE TITLE
MS Surface Pro 9 Win 10 file not found

### ISSUE THREAD

**Issue by cjmx2000** [maurice-daly/DriverAutomationTool#487]
download from https://download.microsoft.com/download/e/9/a/e9a35d1b-7b4c-48d4-8619-5301d4c09f65/SurfacePro9_Win10_19044_22.101.15507.0.msi

file not found.  Any help would be appreciated.

**Comment by bobojobaxter** [comment]
Same. I checked MS and it shows the files: 
SurfacePro9_Win10_19044_23.044.40192.0
SurfacePro9_Win11_22621_23.044.40352.0

**Comment by ajn142attamu** [comment]
Hey, I think my fork might fix the Surface driver issue, we had the same issue with Windows 11 22H2. I haven’t figured out compiling the .exe or .msi like the repo owner, so you have to run the .ps1 directly from an Admin Powershell window.

https://github.com/ajn142attamu/DriverAutomationTool/blob/master/Current%20Branch/7.2.2/DriverAutomationTool.ps1

Let me know if that helps and I can see about putting in a PR.

**Comment by cjmx2000** [comment]
> Hey, I think my fork might fix the Surface driver issue, we had the same issue with Windows 11 22H2. I haven’t figured out compiling the .exe or .msi like the repo owner, so you have to run the .ps1 directly from an Admin Powershell window.
> 
> https://github.com/ajn142attamu/DriverAutomationTool/blob/master/Current%20Branch/7.2.2/DriverAutomationTool.ps1
> 
> Let me know if that helps and I can see about putting in a PR.

This works correctly!  thanks!

**Comment by ChemKroonri** [comment]
I have your ps1  a try and it does not solve the issue. it looks like the Dev needs to update the json file with new information.
because the links in there for the SP9 give me a 404 error

https://[raw.githubusercontent.com/maurice-daly/DriverAutomationTool/master/Data/OSDCatalogMicrosoftDriverPack.json

**Comment by ajn142attamu** [comment]
> I have your ps1 a try and it does not solve the issue. it looks like the Dev needs to update the json file with new information. because the links in there for the SP9 give me a 404 error
> 
> https://raw.githubusercontent.com/maurice-daly/DriverAutomationTool/master/Data/OSDCatalogMicrosoftDriverPack.json

You may find the master branch (or possibly the SelfHostMSCatalog branch) on my repo more helpful. In short, my patch fixed the issue at-the-time, but like you found, DAT also depends on a catalog, and if the catalog is outdated, you're hosed. In the time since I posted my initial patch, both Dell and MS have changed their download site format, and MS has released new driver packs as well. From what I worked out, Maurice uses a tool called OSDeploy/OSD to generate the catalogs, and then hosts his own catalog files in this repo. My main branch changed that, and points at the OSDeploy repo instead, where those catalogs are updated ~1x/week to 1x/month.

**Comment by ChemKroonri** [comment]
Thanks I got is to download with your selfhosted ps1 file. going to test it tomorrow

**Comment by ajn142attamu** [comment]
Sounds good. My self-hosted files *may* be out of date, if I have time today I’ll commit new catalogs and stuff.
________________________________

**Comment by cjmx2000** [comment]
@ajn142attamu are you able to update the catalog again?  Thanks!

**Comment by GeoSimos** [comment]
Workaround for the catalog here: https://www.microsoft.com/en-us/download/details.aspx?id=57439

**Comment by ajn142attamu** [comment]
@cjmx2000 I'm sorry, I've been stuck buried under other issues, but I may revisit soon. In the meantime, if you change line 26 of your OEMLinks.xml to the value below it *should* fix things.

```
		<Link Type="JSONSource" URL="https://raw.githubusercontent.com/OSDeploy/OSD/master/Catalogs/MicrosoftDriverPackCatalog.json"></Link>
```

**Comment by GeoSimos** [comment]
> @cjmx2000 I'm sorry, I've been stuck buried under other problems at work, but I may revisit soon. In the meantime, if you change line 26 of your OEMLinks.xml to the value below it _should_ fix things.
> 
> ```
> 		<Link Type="JSONSource" URL="https://raw.githubusercontent.com/OSDeploy/OSD/master/Catalogs/MicrosoftDriverPackCatalog.json"></Link>
> ```

This catalog is current?

**Comment by cjmx2000** [comment]
@ajn142attamu 

this is the link I have already
		<Link Type="JSONSource" URL="https://raw.githubusercontent.com/OSDeploy/OSD/master/Catalogs/MicrosoftDriverPackCatalog.json"></Link>

It tries to download from here and never connects

**Comment by ajn142attamu** [comment]
> This catalog is current?

Yes @GeoSimos updated as of 2 days ago. https://github.com/OSDeploy/OSD/blob/master/Catalogs/MicrosoftDriverPackCatalog.json

**Comment by ajn142attamu** [comment]
> https://download.microsoft.com/download/e/9/a/e9a35d1b-7b4c-48d4-8619-5301d4c09f65/SurfacePro9_Win10_19044_22.101.15507.0.msi

Really odd, because if you look at the online catalog, it should be trying to pull the link below for any Surface Pro 9:

https://download.microsoft.com/download/e/9/a/e9a35d1b-7b4c-48d4-8619-5301d4c09f65/SurfacePro9_Win10_19044_24.012.27594.0.msi

**Comment by GeoSimos** [comment]
> https://raw.githubusercontent.com/OSDeploy/OSD/master/Catalogs/MicrosoftDriverPackCatalog.json

This is not the same filename, have you cleaned any cached files from the tool and restart it?

**Comment by cjmx2000** [comment]
@GeoSimos this is the correct link per @ajn142attamu  https://raw.githubusercontent.com/OSDeploy/OSD/master/Catalogs/MicrosoftDriverPackCatalog.json

removed all files and started over.  Success!  thanks for the assistance, I should have started with that. 
- DriverPackage: Drivers copied to - S:\Driver Packages\Microsoft\Surface_Pro_9_for_Business_2038\Windows11-22H2-x64-24.012.27601.0\StandardPkg\

**Comment by GeoSimos** [comment]
> @GeoSimos this is the correct link per @ajn142attamu https://raw.githubusercontent.com/OSDeploy/OSD/master/Catalogs/MicrosoftDriverPackCatalog.json
> 
> removed all files and started over. Success! thanks for the assistance, I should have started with that.
> 
> * DriverPackage: Drivers copied to - S:\Driver Packages\Microsoft\Surface_Pro_9_for_Business_2038\Windows11-22H2-x64-24.012.27601.0\StandardPkg\

Glad it worked for you, have pulled my hair some times and this was the solution ;-)
